components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/mqtt/MqttConnectionFactory.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.inbound.endpoint.protocol.mqtt;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.eclipse.paho.client.mqttv3.MqttAsyncClient;
+import org.eclipse.paho.client.mqttv3.MqttClient;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.paho.client.mqttv3.persist.MqttDefaultFilePersistence;
+
+import java.util.Hashtable;
+import java.util.Properties;
+
+/**
+ * MQTT factory, return instances of asynchronous and synchronous clients
+ */
+public class MqttConnectionFactory {
+
+    private static final Log log = LogFactory.getLog(MqttConnectionFactory.class);
+
+    private String factoryName;
+    private int retryInterval = 10000;
+    private int retryCount = 3;
+    private Hashtable<String, String> parameters = new Hashtable<String, String>();
+
+    public MqttConnectionFactory(Properties passedInParameter) {
+
+        this.factoryName = passedInParameter.getProperty(MqttConstants.PARAM_MQTT_CONFAC);
+
+        try {
```

Comment:
When a invalid or no parameter specified it'll only print a log and proceed? Have you tested such scenarios.